handlers/canvas_handler.py

```diff
@@ -130,7 +136,8 @@ def track_course(self, course_ids_str: Tuple[str, ...]):
         course_ids = self._ids_converter(course_ids_str)
         c_ids = {c.id for c in self.courses}
 
-        self.courses.extend(self.get_course(i) for i in course_ids if i not in c_ids)
+        new_courses = [self.get_course(i) for i in course_ids if i not in c_ids]
```

Comment:
This should probably be a tuple since it is not modified, only iterated through which is slightly more efficient with a tuple over a list. If you do change it, be sure it's actually a tuple and not a generator since a generator would have to run twice in this situation.